src/app/core/data/array-move-change-analyzer.service.ts

```diff
@@ -0,0 +1,37 @@
+import { MoveOperation } from 'fast-json-patch/lib/core';
+import { Injectable } from '@angular/core';
+import { moveItemInArray } from '@angular/cdk/drag-drop';
+import { hasValue } from '../../shared/empty.util';
+
+/**
+ * A class to determine move operations between two arrays
+ */
+@Injectable()
+export class ArrayMoveChangeAnalyzer<T> {
```

Comment:
I think this class is missing specs?  Unless I overlooked them.   It'd be important to add specs here to ensure it is generating the correct MoveOperation when diffing Arrays